main.cpp

```diff
@@ -636,13 +589,36 @@ int main(int argc, char *argv[])
     parser->insertOption(17,"--http-proxy 1");
     parser->insertOption(18,"--socks5-proxy 1");
 
+    parser->insertOption(19,"--no-stats"); //don't collect and send anonymous statistics
+
 
     //...............
 
     parser->parse(opts);
 
+    if(!parser->isParamExist("no-stats")){
+        // Application instance definition
 
-    int ret;
+        QString AAID=getAIID();
+        QString OS=getOS();
+
+        utsname u;
+        uname(&u);
+
+        QString PLATFORM=u.machine;
+        QString DISTR=QString(u.sysname)+" "+QString(u.release);
+
+
+        MSRequest* req=new MSRequest();
```

Comment:
Yes you right. I'm forgot make delete(req) after request sending. But since it occurs in main() function memory leak will very small. Many thanks